relay-cabi/src/processing.rs

```diff
@@ -9,29 +9,149 @@ use std::os::raw::c_char;
+    /// information should the event payload contain no such data. If no client hints are present,
+    /// normalization falls back to the user agent.
+    pub client_hints: ClientHints<String>,
+
+    /// The time at which the event was received in this Relay.
+    ///
+    /// This timestamp is persisted into the event payload.
+    pub received_at: Option<DateTime<Utc>>,
+
+    /// The time at which the event was sent by the client.
+    ///
+    /// The difference between this and the `received_at` timestamps is used for clock drift
+    /// correction, should a significant difference be detected.
+    pub sent_at: Option<DateTime<Utc>>,
+
+    /// The maximum amount of seconds an event can be predated into the future.
+    ///
+    /// If the event's timestamp lies further into the future, the received timestamp is assumed.
+    pub max_secs_in_future: Option<i64>,
+
+    /// The maximum amount of seconds an event can be dated in the past.
+    ///
+    /// If the event's timestamp is older, the received timestamp is assumed.
+    pub max_secs_in_past: Option<i64>,
+
+    /// When `Some(true)`, individual parts of the event payload is trimmed to a maximum size.
+    ///
+    /// See the event schema for size declarations.
+    pub enable_trimming: Option<bool>,
+
+    /// When `Some(true)`, it is assumed that the event has been normalized before.
+    ///
+    /// This disables certain normalizations, especially all that are not idempotent. The
+    /// renormalize mode is intended for the use in the processing pipeline, so an event modified
+    /// during ingestion can be validated against the schema and large data can be trimmed. However,
+    /// advanced normalizations such as inferring contexts or clock drift correction are disabled.
+    ///
+    /// `None` equals to `false`.
+    pub is_renormalize: Option<bool>,
+
+    /// Overrides the default flag for other removal.
+    pub remove_other: Option<bool>,
+
+    /// When `Some(true)`, context information is extracted from the user agent.
+    pub normalize_user_agent: Option<bool>,
+
+    /// Emit breakdowns based on given configuration.
+    pub breakdowns: Option<BreakdownsConfig>,
+
+    /// The SDK's sample rate as communicated via envelope headers.
+    ///
+    /// It is persisted into the event payload.
+    pub client_sample_rate: Option<f64>,
+
+    /// The identifier of the Replay running while this event was created.
+    ///
+    /// It is persisted into the event payload for correlation.
+    pub replay_id: Option<Uuid>,
+
+    /// Controls whether spans should be normalized (e.g. normalizing the exclusive time).
+    ///
+    /// To normalize spans in [`normalize_event`], `is_renormalize` must
+    /// be disabled _and_ `normalize_spans` enabled.
+    pub normalize_spans: bool,
+}
+
+impl StoreConfig {
+    /// Helper method to parse *mut StoreConfig -> &StoreConfig
+    pub fn this(&self) -> &Self {
```

Comment:
```suggestion
   fn this(&self) -> &Self {
```

Looks like this doesn't need to be public